### ISSUE TITLE
SWALP Example

### ISSUE THREAD

**Issue by andwgu** [Tiiiger/QPyTorch#52]
Beginner question: Why do we not need to quantize the non-convolutional, non-linear layers in the VGG network (e.g. `BatchNorm2d`, `ReLU`, `Dropout`, etc.)? In other words, why do we not pass all of the layer types into `layer_types` to ensure that every tensor used is quantized?

https://github.com/Tiiiger/QPyTorch/blob/176bf2a4b02021b116991d9b2f4d920861725f26/examples/SWALP/train.py#L118-L125

If I am understanding correctly, SWALP involves quantizing everything. Is the reason that those other layers do not have parameters, so the forward/backward pass through them will preserve the low-precision? (Although, `BatchNorm2d` has parameters for the affine transformation.)

**Comment by Tiiiger** [comment]
hi, 

I think you are correct: the nonlinearites don't change the precision of the underlying numbers

BatchNorm can be seen as fused with the following linear layer in terms of precision. So they are quantized "together" but they same quantizer module after it.
